### ISSUE TITLE
Slow performance on rule `@pandacss/no-config-function-in-source`

### ISSUE THREAD

**Issue by FranciscoKloganB** [chakra-ui/eslint-plugin-panda#67]
Hey folks, thank you for the Eslint plugin. Recently I noticed some sluginesh on my project both during pre-commits and also after file changes. My CPU would start picking up. I wouldn't say this eslint plugin is the sole reason, but I ran into this during some of my diagnose processes:

Running over 467 files (at most) and running.

```bash
TIMING=1 pnpm eslint .storybook cypress mocks src tests themes
```

Yields the folloowing results. PandaCSS eslint plugin is a real heavy hitter, but the one rule that does concern me is `no-config-function-in-source`. Maybe you guys can look into optimizing this somehow?

```bash
Rule                                      | Time (ms) | Relative
:-----------------------------------------|----------:|--------:
@pandacss/no-config-function-in-source    | 11619.024 |    64.0%
@typescript-eslint/promise-function-async |  2058.247 |    11.3%
@pandacss/file-not-included               |   849.771 |     4.7%
@pandacss/no-hardcoded-color              |   602.716 |     3.3%
@pandacss/no-unsafe-token-fn-usage        |   563.741 |     3.1%
@pandacss/no-invalid-token-paths          |   554.260 |     3.1%
@pandacss/no-dynamic-styling              |   451.480 |     2.5%
@pandacss/no-property-renaming            |   362.736 |     2.0%
@typescript-eslint/naming-convention      |   307.606 |     1.7%
import/no-duplicates                      |   278.164 |     1.5%
```

**Comment by anubra266** [comment]
Hi @FranciscoKloganB Is there any improvements in the latest version

**Comment by FranciscoKloganB** [comment]
Hey Hey. Much better. :)

```bash
Rule                                      | Time (ms) | Relative
:-----------------------------------------|----------:|--------:
@typescript-eslint/promise-function-async |  2774.613 |    39.4%
@pandacss/file-not-included               |   952.042 |    13.5%
@pandacss/no-hardcoded-color              |   562.673 |     8.0%
@pandacss/no-unsafe-token-fn-usage        |   537.587 |     7.6%
@pandacss/no-invalid-token-paths          |   525.060 |     7.5%
@pandacss/no-dynamic-styling              |   403.756 |     5.7%
@pandacss/no-property-renaming            |   364.437 |     5.2%
import/no-duplicates                      |   277.866 |     3.9%
@typescript-eslint/naming-convention      |   236.710 |     3.4%
vitest/expect-expect                      |    55.430 |     0.8%
```
